### DISCUSSION TITLE
✅ How do I use private / public SSH keys with WebSSH on iPad ?

### DISCUSSION THREAD

**Discussion by GerardWassink** [isontheline/pro.webssh.net#369] [Q&A] [answered]
On my local machine I use ssh-genkey to generate a key-pair to be able to connect to my cloud server w/o password.

How do I do that with WebSSH on my iPad?

Gerard

**Comment by isontheline** [comment, marked as answer]
Hello @GerardWassink 👋

Just add your private key part on your connection :
![image](https://user-images.githubusercontent.com/44212923/127917403-1f7000da-535c-4859-a1d0-c54dd470c006.png)
![image](https://user-images.githubusercontent.com/44212923/127917469-79cda67e-19d5-41e7-abf5-c352063f45a0.png)

Have a nice day 👋

**Reply by GerardWassink** [reply]
After some fiddling I got it to work. Used the private key I generated on my Win 10 Machine and transferred the contents to my iPad, followed your recipe above. Now I can connect without using my 2FA. 
Thanks!
Gerard

**Reply by isontheline** [reply]
You're Welcome!

Thank you a lot for your feedback.

Have a nice day ☀️
